chapter04_convolutional-neural-networks/kaggle-gluon-cifar10.md

```diff
@@ -0,0 +1,412 @@
+# 实战Kaggle比赛——使用Gluon对原始图像文件分类（CIFAR-10）
+
+
+我们在[监督学习中的一章](../chapter02_supervised-learning/kaggle-gluon-kfold.md)里，以[房价预测问题](https://www.kaggle.com/c/house-prices-advanced-regression-techniques)为例，介绍了如何使用``Gluon``来实战[Kaggle比赛](https://www.kaggle.com)。社区中的很多小伙伴实践了房
+价预测问题并[分享了自己的成绩和方法](https://discuss.gluon.ai/t/topic/1039)。有小伙伴[反馈希望增加Kaggle比赛章节](https://discuss.gluon.ai/t/topic/1341)并[反馈希望多从直接与数据打交道中获得灵感](https://discuss.gluon.ai/t/topic/1519)。还有小伙伴[反馈希望提供对原始图像分类的例子](https://discuss.gluon.ai/t/topic/499)，
+例如输入格式是jpg或png而非教程中封装好的图像数据，这样甚至可能[更好地管理手机图片](https://discuss.gluon.ai/t/topic/1372)。
+
+
+有鉴于大家的反馈，我们在本章中选择了Kaggle中著名的[CIFAR-10原始图像分类问题](https://www.kaggle.com/c/cifar-10)。我们以该问题为例，为大家提供使用`Gluon`对原始图像文件进行分类的示例代码。
+
+本章的示例代码主要分五部分：
+
+* 整理原始数据集；
+* 使用`Gluon`读取整理后的数据集；
+* 设计模型；
+* 使用部分训练数据训练模型并以其它训练数据为验证集调参；
+* 使用全部训练数据训练模型并对测试数据分类。
+
+
+需要注意的是，本章仅提供一些基本实战流程供大家参考。对于数据的预处理、模型的设计和参数的选择等，我们特意提供一个仅供示例的版本。希望大家通过动手实战、仔细观察实
+验现象、认真分析实验结果并不断调整方法，从而不断提升成绩。
+
+计算机视觉一直是深度学习的主战场，请
+
+> Get your hands dirty。
+
+
+
+
+## Kaggle中的CIFAR-10原始图像分类问题
+
+[Kaggle](https://www.kaggle.com)是一个著名的供机器学习爱好者交流的平台。为了便于提交结果，请大家注册[Kaggle](https://www.kaggle.com)账号。然后请大家先点击[CIFAR-10原始图像分类问题](https://www.kaggle.com/c/cifar-10)了解有关本次比赛的信息。
+
+![](../img/kaggle_cifar10.png)
+
+
+
+## 整理原始数据集
+
+比赛数据分为训练数据集和测试数据集。训练集包含5万张图片。测试集包含30万张图片：其中有1万张图片用来计分，但为了防止人工标注测试集，里面另加了29万张不计分的图片。
+
+两个数据集都是png彩色图片，大小为$32\times 32 \times 3$。训练集一共有10类图片，分别为飞机、汽车、鸟、猫、鹿、狗、青蛙、马、船和卡车。
+
+（那么问题来了，你觉得你用肉眼能把下面100个图片正确分类吗？）
+
+![](../img/cifar10.png)
+
+
+### 下载数据集
+
+
+登录Kaggle后，数据可以从[CIFAR-10原始图像分类问题](https://www.kaggle.com/c/cifar-10)中下载。
+
+* [训练数据集train.7z下载地址](https://www.kaggle.com/c/cifar-10/download/train.7z)
+* [测试数据集test.7z下载地址](https://www.kaggle.com/c/cifar-10/download/test.7z)
+* [训练数据标签trainLabels.csv下载地址](https://www.kaggle.com/c/cifar-10/download/trainLabels.csv)
+
+如果不登录Kaggle，比赛数据也可通过以下地址下载：
+
+* [训练数据集train.7z下载地址](https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/kaggle-cifar10/train.7z)
+* [测试数据集test.7z下载地址](https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/kaggle-cifar10/test.7z)
+* [训练数据标签trainLabels.csv下载地址](https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/kaggle-cifar10/trainLabels.csv)
+
+
+### 解压数据集
+
+训练数据集train.7z和测试数据集test.7z都是压缩格式，下载后请解压缩。假设解压缩后原始数据集的路径如下（data_dir是存放数据的路径，例如'..
+/data/kaggle_cifar10'，'train_dir'和'test_dir'分别是对train.7z和test.7z解压后的文件夹名称，例如'train'和'test'）：
+
+* data_dir/train_dir/1.png
```

Comment:
use code block, and real data position

```
../data/kaggle_cifar10/train_dir/1.png
...
../data/kaggle_cifar10/train_dir/50000.png
```

Reply:
using real data position now, removing verbosity to make it more suitable for markdown block